### ISSUE TITLE
Still new to Structural MRI analysis. Is it possible to use multiple priors?

### ISSUE THREAD

**Issue by birkenstonks** [ANTsX/ANTs#1297]
I'm running 'antsBrainExtraction.sh' and was wondering if I could use the priors for CSF, GM, and WM instead of prioritizing just one. 

I tried including 3 priors as arguments in my command, but in the output it looks like just the first is used.

**Comment by ntustison** [comment]
The only prior used in ``antsBrainExtraction.sh`` is the ``-m`` template probability whole brain mask.  I'm guessing you if your CSF, GM, and WM priors correspond to your brain template, you should be able to add them together to create the input.

**Comment by birkenstonks** [comment]
Ok, how can I add the priors together? If I understand what you're saying, does that mean the template file I'm using with the -e argument is the combination of these different priors?

**Comment by ntustison** [comment]
Please look at [this example](https://github.com/ntustison/antsBrainExtractionExample).

**Comment by birkenstonks** [comment]
Ok I just checked the example and looked at the priors and template. I saw my template and priors are 3D volumes whereas the example has 2D images, but my priors are separated between the different tissue types. Do you have suggestions for how I can combine them into one?

**Comment by cookpa** [comment]
Are you asking how to combine your prior probabilities into a template brain mask for the `-m` option?

You can do that with `ImageMath`. Example:

`ImageMath 3 classLabels.nii.gz MostLikely 0.01 prior*`

You can then binarize this (`ThresholdImage`) and optionally smooth it

**Comment by birkenstonks** [comment]
Yes that's what I'm looking to do! Where is the classLabels.nii.gz file coming from? If I have 3 priors.nii files, can my command look like 'ImageMath 3 priors1.nii priors2.nii priors3.nii MostLikely 0.01 prior*' ?

**Comment by cookpa** [comment]
`classLabels.nii.gz` would be the output.

**Comment by birkenstonks** [comment]
Incredible. Thank you that worked

**Comment by birkenstonks** [comment]
I was able to combine the priors, but the output files seem to just have CSF. The log shows a message that 'Mask not read.  Using the entire image as the mask,' so I'm guessing it may be related to that. Any suggestions how I can fix this? I'm including a screenshot of the priors and of one of the output files  

Another question, @cookpa you mentioned I could smooth the image. How can I do that?
<img width="543" alt="Screen Shot 2022-01-28 at 3 33 11 PM" src="https://user-images.githubusercontent.com/66442660/151617760-03c73dc1-a566-408b-abba-bc423f0d53ac.png">
<img width="501" alt="Screen Shot 2022-01-28 at 3 32 50 PM" src="https://user-images.githubusercontent.com/66442660/151617782-cebb6ae6-775e-4004-bbda-aa50e26fae7d.png">

**Comment by cookpa** [comment]
The first image looks correct. Threshold this:

`ThresholdImage 3 classLabels.nii.gz brainMask.nii.gz 1 Inf 1 0`

If you want to smooth, 

`SmoothImage 3 brainMask.nii.gz 1 brainMaskProb.nii.gz 1 0`

You can use the brainMask or brainMaskProb with `antsBrainExtraction.sh` (-m).

**Comment by birkenstonks** [comment]
These look much better! Thanks so much.

If it's not too much trouble, would you mind explaining the different arguments in the ThresholdImage and SmoothImage commands? I'm mainly curious about the numbers 

<img width="533" alt="Screen Shot 2022-01-30 at 10 49 45 PM" src="https://user-images.githubusercontent.com/66442660/151736558-6a012375-274e-4f38-8592-66de0f6ef83a.png">

**Comment by cookpa** [comment]
Glad it helped. You can see the usage for both programs if you run them with no args.

**Comment by JinDJsuper** [comment]
> Are you asking how to combine your prior probabilities into a template brain mask for the `-m` option?
> 
> You can do that with `ImageMath`. Example:
> 
> `ImageMath 3 classLabels.nii.gz MostLikely 0.01 prior*`
> 
> You can then binarize this (`ThresholdImage`) and optionally smooth it

I need to extract the voxel or volume data of segmented gray and white matter. Can I convert it into a binary file and then extract the data?
I tried the parameter of 0.01, and the resulting image segmentation was quite inaccurate, for example, the gray matter part was greatly enlarged. So I changed it to 0.5. Is this parameter suitable?he image might differ slightly from the probability map, but most of it aligns correctly.
`ImageMath 3 gm.nii.gz MostLikely 0.5 brain10_gm_prior.nii`
Or can I directly extract the voxels from the probability map, such as in this way?
`ImageMath 3 gm total brain10_gm_prior.nii`
I looked at the data, and it seems that the difference isn't significant. first  gm is  41407.6 ，second gm is 40215.5. So, I want to ask, what method is best for extracting the segmented gray and white matter voxels?
